nilearn/plotting/surf_plotting.py

```diff
@@ -486,14 +486,19 @@ def _threshold_and_rescale(data, threshold, vmin, vmax):
     This function thresholds and rescales the provided data.
     """
     data_copy, vmin, vmax = _rescale(data, vmin, vmax)
-    return data_copy, _threshold(data, threshold), vmin, vmax
+    return data_copy, _threshold(data, threshold, vmin), vmin, vmax
 
 
-def _threshold(data, threshold):
+def _threshold(data, threshold, vmin):
     """Thresholds the data."""
     # If no thresholding and nans, filter them out
-    return (np.logical_not(np.isnan(data)) if threshold is None
-            else np.abs(data) >= threshold)
+    if threshold is None:
+        mask = np.logical_not(np.isnan(data))
+    else:
+        mask = np.abs(data) >= threshold
+        if vmin > -threshold:
+            mask = np.logical_and(mask, data >= vmin)
```

Comment:
@michellewang I just have one question left about this addition because the resulting behavior of this change is not clear to me. Can you show the behavior difference here given when the condition is met and when it's not met (an example in the PR is fine)? I'm just wondering if it would be clear to the user how vmin value will affect the thresholding.

Reply:
@ymzayek if `vmin <= -threshold` then the old behaviour (`mask = np.abs(data) >= threshold`) works as expected: the image is clipped at `vmin` and shows data between `-threshold` and `vmin`

```python
from nilearn.datasets import load_sample_motor_activation_image
from nilearn import plotting

img = load_sample_motor_activation_image()
plotting.plot_img_on_surf(
    img,
    colorbar=True,
    symmetric_cbar=False,
    vmin=-3,                          # vmin <= -threshold
    threshold=2,
    cmap='viridis',
)
```

![image](https://github.com/nilearn/nilearn/assets/29051929/bd631771-2b22-4c51-9c85-6e88ec62b72d)

Now that we have `vmin`, the issue with the previous version of `_threshold()` was that it doesn't account for cases where the data range is asymmetric, for example if the user only wants to plot positive values. Below is what the plot looks like with the old `_threshold()` behaviour if `vmin=0` (i.e., `vmin > -threshold`). We can see that the colourbar is correct but the actual image still shows negative values beyond the threshold, which is unexpected.

![image](https://github.com/nilearn/nilearn/assets/29051929/978c1d19-fecc-437b-aba8-f2a137711fb8)

Updating the mask based on `vmin` if `vmin > -threshold` fixes this:

![image](https://github.com/nilearn/nilearn/assets/29051929/9dcd22a2-b01e-46a5-b3b9-07ecadf0dca8)

And now I just realized that we would have a similar problem when `vmax < threshold` (for example if the user only wanted to plot negative values, with something like `vmax = 0` and `threshold = 2`), so I will add another part to `_threshold()` so that it also updates the mask in that situation.

Let me know if this makes sense and/or if there are edge cases that I missed and/or we should clarify the new behaviour in the docstring.

Reply:
Very clear thanks! I think with the addition of handling vmax and a green CI we should be good to go.